### PR TITLE
feat(equalTo): Add equalTo validator

### LINKED ISSUE #58: New feature request: "equalsTo" validator
When I validate registration form, I want to validate password confirmation with schema like this:
```js
let model = obey.model({
	/* Unrelated fields omitted */
	password: {
		type: 'string',
		min: 8
	},
	passwordConfirm: {
		type: 'string',
		equalsTo: 'password'
	}
});
```

### PR DESCRIPTION
**Pull request by devsli**
Fixes #58

### CODE REVIEW

**Comment by Fluidbyte** [comment]
Initial run-through looks really good. One thing I'd suggest would be changing the property name from `equalsTo` to just `equalTo`. I know it's kind of picky, but I feel like it's more "natural" when considering other properties (`empty`, `allow`, `required`...) and may be a little easier to remember.

**Comment by devsli** [comment]
Replaced that commit with new one, now it named `equalTo`

File: src/lib/validators.js [hunk 1]
```diff
@@ -100,6 +100,23 @@ const validators = {
...
+   * @param {string} key The key name of the property
+   * @param {Array<{type: string, sub: string|number, key: string, value: *, message: string}>} errors An error array
+   * to which any additional error objects will be added
+   * @param {Object} data The full initial data object
+   */
+  equalTo: function(def, value, key, errors, data) {
+    const type = 'equalTo'
+    const sub = def.equalTo
+    if (dot.pick(sub, data) !== value) {
+      errors.push({ type, sub, key, value, message: `Value doesn\'t match '${sub}' value`})
```
Comment (Fluidbyte) [inline src/lib/validators.js]: Can you change this to the instructional `Value must match ${sub} value ${dot.pick(sub, data)}`

Comment (Fluidbyte) [inline src/lib/validators.js]: Want it to match convention, see [max validator](https://github.com/TechnologyAdvice/obey/blob/master/src/lib/validators.js#L67)

Comment (devsli) [inline src/lib/validators.js]: So the password will be visible as a part of error message? What about `Value must match ${sub} value`?

Comment (Fluidbyte) [inline src/lib/validators.js]: Oh, good call, let's just change it to `Value must match ${sub} value}`

Comment (davezuko) [inline src/lib/validators.js]: Oh hah, those comments never came up until I refreshed. Oh well...

**Comment by devsli** [comment]
Just updated, now error message is `Value must match ${sub} value`

**Pull request merged**